contracts/extensions/CoinMachine.sol

```diff
@@ -144,32 +158,47 @@ contract CoinMachine is ColonyExtension {
   function buyTokens(uint256 _numTokens) public payable notDeprecated {
     updatePeriod();
 
-    uint256 numTokens = min(min(_numTokens, maxPerPeriod - activeSold), tokensToSell);
+    require(
+      whitelist == address(0x0) || Whitelist(whitelist).approved(msg.sender),
```

Comment:
Hmm, good question. What do you think? I'd be inclined to say re-deployed, because the case where someone accidentally uninstalls the whitelist and opens up coin machine to the public is worse than one where coin machine must be explicitly deployed publically.

Reply:
I agree that shouldn't be the failure state; I was thinking a solution should be to have a function that can set the whitelist.